### ISSUE TITLE
add permission per world and permission time expired

### ISSUE THREAD

**Issue by padreon** [CortexPE/Hierarchy#6]
add permission per world and permission time expired

**Comment by CortexPE** [comment]
What for?

**Comment by Thunder33345** [comment]
per world for "per world" uses?
say to allow doing 1 thing only in 1 world, like allowing factions commands in factions worlds, and plots command in plot worlds

for expiry maybe that should be an extension plugin of a plugin to not bloat main plugin
uses would be something like a subscription rank

**Comment by artulloss** [comment]
I disagree with per world permissions because it's overly complicating things for a poor use case; if you have the option of mysql then why put everything on one server. I think both of these ideas should be extension plugins and left out of the main because it really depends on how a server is setup.

**Comment by CortexPE** [comment]
yeah I don't think the majority of users have the need for multi-world permissions

**Comment by Thunder33345** [comment]
I agree it's rather niche i am not defending it but just thinking of possible use cases\

the per world might be a mess, not sure if implementing on a plugin level would make it even messier 
the timed permission should be a bliss to be done as an plugin ext, just add/remove when time comes

**Comment by CortexPE** [comment]
The per-world one would be messy...

But it could be theoretically done by creating in-memory / temporary roles for the roles that have per-world permissions... Then add & remove those from members that have the original roles when they join & move levels.

Then for the member perms, Same concept... Add & remove perms on join & level transfer.

The commands would be separate.

Per-world extension:
```
/wordhrk grantperm <world> <member|role> <permission>
/wordhrk removeperm <world> <member|role> <permission>
/wordhrk denyperm <world> <member|role> <permission>
```

Temporary roles & perms extension:
```
/temphrk grantperm <member|role> <permission> <expiryDate>
/temphrk denyperm <member|role> <permission> <expiryDate>
/temphrk removeperm <member|role> <permission>
```

**Comment by LeoBdt** [comment]
It'd be cool if you add temporary roles with date or amount of hours/minutes. 

For example : If a player vote i want to give him fly permission for 15 minutes.

Regards.

**Comment by artulloss** [comment]
I'm planning on making an extension to support timed ranks in the coming days, and actually for the same reason you just said.

**Comment by LeoBdt** [comment]
You're on the good way for being better then PurePerms !

**Comment by raymond-itsec** [comment]
I have a few use cases for multi world permissions with multiple minigame worlds.

**Comment by suerion** [comment]
I had 2 servers, one for survival and one for creativ, but for any games on the survival server, i need to add some roles for multiworld, i think it is for a lot of reasons needed, to at multiworld support. Thanks for your working

The games is made for an player allway an new map. Skyblock is a little bit confusiong, but with it, ther are a lot of multiworlds

**Comment by suerion** [comment]
@CortexPE is there a timeline, when you will add multiworld support?

**Comment by suerion** [comment]
Per Multiworld, it should create also permissions for prefix worlds like Skyblock

skyblock generates allways a new World with `sb-`

it would famous if you could support it
